### PR TITLE
common: docker/Dockerfile and scripts/README.md fixes and improvements

### PR DESCRIPTION
**Pull request by janekmi**
Requires:

- [ ] #5

---
This change is [Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+after that, the flow of the admin steps is:
+
+## foreach hackathon:
+
+1. create a local clone of the hackathon repo in `/home/hackathon`, e.g.:
+
+```sh
+cd /home/hackathon
+git clone https://githib.com/pmemhackathon/2019-07-07
```
Comment (ldorau) [inline scripts/README.md]: github

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+2. create users, e.g.:
+
+```sh
+# creates all 200 accounts
+create_pmemusers 1 200 2019-07-07
```
Comment (ldorau) [inline scripts/README.md]: What directory should be this command run from ?

File: scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+## foreach session with same users (or on system boot)
+
+start the webhackathon daemon as root
+
+```sh
+cd /home/hackathon
+./webhackathon 2019-07-07 &
+enable_pmemusers 1 100 todayspasswd
```
Comment (ldorau) [inline scripts/README.md]: what does " 1 100" mean ?

Comment (ldorau) [inline scripts/README.md]: What directory should be this command run from ?

**Review by ldorau — CHANGES_REQUESTED** [review]

File: scripts/README.md (hunk 1 above)
Comment (janekmi) [reply scripts/README.md]: Done. #5

File: scripts/README.md (hunk 2 above)
Comment (janekmi) [reply scripts/README.md]: Done.

File: scripts/README.md (hunk 3 above)
Comment (janekmi) [reply scripts/README.md]: Done.

Comment (janekmi) [reply scripts/README.md]: Done.

**Review by ldorau — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi)

___
*[scripts/README.md, line 64 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcePef06tObMCfSb-1:-MkcePegCdkQdgQqGVxZ:b-4hupxx) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L64)):*
> ```
> In case of any errors, you can investigate the reasons for the failure 
> ```

1) `of the failure`
2) maybe change to singular?
`In case of any error, you can investigate the reason of the failure ...`
___
*[scripts/README.md, line 66 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcetUu1zCGQROtbVRP:-MkcetUu1zCGQROtbVRQ:b-q1z05n) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L66)):*
> ```
> Enter the web browser (e.g. [127.0.0.1](http://127.0.0.1)) and
> ```

`Launch a web browser and navigate to  ([http://127.0.0.1](http://127.0.0.1))  and`

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @ldorau)

___
*[scripts/README.md, line 64 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcePef06tObMCfSb-1:-MkcwbP79SnaFT5OFU2u:b-896fix) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L64)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

> ```
> In case of any errors, you can investigate the reasons for the failure 
> ```

1) `of the failure`
2) maybe change to singular?
`In case of any error, you can investigate the reason of the failure ...`
</blockquote></details>

Done.
___
*[scripts/README.md, line 66 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcetUu1zCGQROtbVRP:-MkcxJgA76Us6IrTX9-A:b-896fix) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L66)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

> ```
> Enter the web browser (e.g. [127.0.0.1](http://127.0.0.1)) and
> ```

`Launch a web browser and navigate to  ([http://127.0.0.1](http://127.0.0.1))  and`
</blockquote></details>

Done.

**Review by ldorau — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[scripts/README.md, line 64 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcePef06tObMCfSb-1:-Mkd-1i54C9uXEOx56oU:bayvp5e) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L64)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

"failure" is used two times now and it does not sound good...
Maybe `the reason for that ... ` ?

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: 1 of 3 files reviewed, 1 unresolved discussion (waiting on @ldorau)

___
*[scripts/README.md, line 64 at r3](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-MkcePef06tObMCfSb-1:-MkgUFHs3zCjTfmLYzdn:b-896fix) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/5cea784be2f9c230ebb3db52ee45dbffaf3ee4f8/scripts/README.md#L64)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

"failure" is used two times now and it does not sound good...
Maybe `the reason for that ... ` ?
</blockquote></details>

Done.

**Review by ldorau — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

**Review by ldorau — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-:-MkgVEZN3PYVcBOjcb7k:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @janekmi)

**Review by lukaszstolarczuk — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi)

___
*[docker/Dockerfile, line 7 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-Mkr841e3ZijffwmBtPT:-Mkr841e3ZijffwmBtPU:b8hp51o) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/docker/Dockerfile#L7)):*
> ```Dockerfile
> #       the image:
> #
> # $ sudo docker build -t pmemhackathon/pmemfc30:09 -f ./docker/Dockerfile ./docker/
> ```

1. why `pmemfc30`? - just `fc30` should be fine
2. why `:09`? if you "need" a tag, perhaps `:latest`
___
*[scripts/README.md, line 3 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-Mkr8b7J1soHwflzxQFT:-Mkr8b7J1soHwflzxQFU:bpl4yqv) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L3)):*
> ```Markdown
> # admin scripts
> 
> **Note**: Please replace '2019-07-07' with the name of the repository.
> ```

perhaps update this to the current name already

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi and @lukaszstolarczuk)

___
*[docker/Dockerfile, line 7 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-Mkr841e3ZijffwmBtPT:-MlFbV-994_K-ZKs3ymr:b3dd8jq) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/docker/Dockerfile#L7)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

1. why `pmemfc30`? - just `fc30` should be fine
2. why `:09`? if you "need" a tag, perhaps `:latest`
</blockquote></details>

It is not my idea. ;-)

Ref: https://github.com/pmemhackathon/2021-04-21.web/blob/master/scripts/enable_pmemusers#L35

**Review by grom72 — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6)* status: all files reviewed, 8 unresolved discussions (waiting on @janekmi and @lukaszstolarczuk)

___
*[`scripts/README.md` line 3 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGgqEs38LE08KSveWV:-NBGgqEs38LE08KSveWW:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L3)):*
> ```Markdown
> # admin scripts
> 
> **Note**: Please replace '2019-07-07' with the name of the repository.
> ```





_Suggestion:_
```Markdown
2021-04-21
```
___
*[`scripts/README.md` line 26 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGgjb75MoafIMJ2bAr:-NBGgjb75MoafIMJ2bAs:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L26)):*
> ```Markdown
> ```sh
> cd /home/hackathon
> git clone https://github.com/pmemhackathon/2019-07-07
> ```





_Suggestion:_
```Markdown
2021-04-21
```
___
*[`scripts/README.md` line 38 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGggEC3YUaPX71N-k4:-NBGggEC3YUaPX71N-k5:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L38)):*
> ```Markdown
> ```sh
> # creates all 200 accounts (run from the top of the source tree; as root)
> scripts/create_pmemusers 1 200 2019-07-07
> ```





_Suggestion:_
```Markdown
2021-04-21
```
___
*[`scripts/README.md` line 61 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGgckRAUd2QJly8dkK:-NBGgckRAUd2QJly8dkL:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L61)):*
> ```Markdown
> ```sh
> cd /home/hackathon
> ./webhackathon 2019-07-07 &
> ```





_Suggestion:_
```Markdown
2021-04-21
```
___
*[`scripts/README.md` line 94 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGgYfvCxabzgOjULuM:-NBGgYfw5OWr4v1BiO4T:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L94)):*
> ```Markdown
> When making changes to:
> - `/home/hackathon/templates/examples/*/*.tmpl` files you have to restart the `webhackathon` daemon
> - `/home/hackathon/2019-07-07/*` files you have to:
> ```





_Suggestion:_
```Markdown
2021-04-21
```
___
*[`scripts/README.md` line 103 at r5](https://reviewable.io/reviews/pmemhackathon/2021-04-21.web/6#-NBGgT6mCD3rFQLr2beZ:-NBGgT6mCD3rFQLr2be_:undefined) ([raw file](https://github.com/pmemhackathon/2021-04-21.web/blob/633c91accc972e9ce9ce4135467b5bce32addcd2/scripts/README.md#L103)):*
> ```Markdown
> # and recreate all users
> ./scripts/delete_pmemusers
> create_pmemusers 1 200 2019-07-07
> ```





_Suggestion:_
```Markdown
2021-04-21
```

**Comment by ldorau** [comment]
@igchor Could you move this PR to the new repo or close it?

**Comment by lukaszstolarczuk** [comment]
@ldorau, I'll move this PR (and resolve existing issues) when https://github.com/pmemhackathon/hackathon.web/pull/4 is closed

**Comment by lukaszstolarczuk** [comment]
continued in https://github.com/pmemhackathon/hackathon.web/pull/6

**Pull request closed without merging**